### ISSUE TITLE
Minimum dimensions of lines and areas

### ISSUE THREAD

**Issue by ollesmaps** [OpenOrienteering/mapper#1021]
It would be cool if minimum dimensions for areas and lines could be predefined. 

For example ISOM2017 minimum dimension for 406 Vegetation, slow running is 225 m2. If this area symbol is drawn and is bellow limit it could be somehow highlighted it is bellow minimum dimensions. The highlight changes only when area is over minimum dimension.

Ideas for highligting: 
- the thin red line that is around objects that are being drawn shall have a different colour as long as the area /length is bellow limit.
- other?

The minimum dimension could be built in within each symbol definition. With that it could be possible to perform more easily an overall checkup of finished map. Something like http://oapp.se/Applikationer/CheckOMap.html

**Comment by dg0yt** [comment]
Do you sometimes use the desktop version? We already do have minimal dimensions. However, they are only used in the "measurement" window at the moment.

The strict way of enforcement would be to NOT show the object (apart from placeholder in a different color). I can hear the outcry when you cannot draw a "narrow ride" below minimal length - people take for granted what they are used to from other products ;-)

**Comment by ollesmaps** [comment]
You are right (now I see it on desktop, most of the time I use the app ) there is a minimum size option. And it shows up when measured. That is great. 

I agree a strict enforcement would be too strict. I would prefer a highlight that shows up already during drawing. This way you can adjust your mapping straight away and do not have to make time costly check ups at the end.

**Comment by lpechacek** [comment]
I myself would be happy for an automated checker checking minimum lengths, areas and gaps. Given enough programmers, the first two are trivial to implement. For the third one I do have some idea. Code contributors wanted! :)

@ollesmaps Another possibility would be to persuade CheckOMap author to extend the tool with OO Mapper format support.

**Comment by dg0yt** [comment]
Another option would be to add this as a special feature/keyword to the "Find" dialog. This gives you both "Find all" and step-by-step.

**Comment by dg0yt** [comment]
Previous tickets: #35 Problem widget, #302 Map report and problem browser tool, #651"Minimal length" of linear symbols.

**Comment by ollesmaps** [comment]
Relates to: http://orienteering.org/new-control-process-for-mapping-for-iof-events/

**Comment by Zerbembasqwibo** [comment]
Related, CheckOmap. http://oapp.se/Applikationer/CheckOMap.html (use translation service for non-Swedish speakers)

It check almost all rules from ISOM (2000/2017), given in numbers but also those extracted from text.

I think it's some thousand rules actually. Is it possible to reuse that big job in some way?

**Comment by ollesmaps** [comment]
Or it can change symbol icon background color when object is being edited - if under limit highlighted color, if ok as usuall. 
![color_symbol](https://user-images.githubusercontent.com/20535899/33836133-d8924fd6-de88-11e7-8408-5a7fc83fcb81.jpg)

**Comment by mlerjen** [comment]
Related: https://github.com/OpenOrienteering/mapper/issues/610

**Comment by ollesmaps** [comment]
At some point I noticed the Measurement tool had built-in under minimum sizes feature. I can not see it there any more (now 0.9.1). That was good and simple solution!

**Comment by lpechacek** [comment]
AFAICT, the function is still there.

However, the map symbol definitions must contain the minimum dimension numbers. For instance, if you open an .ocd file, the minimum size checking won't trigger as the fields are not filled in. Would that be an explanation to what you are seeing?

**Comment by ollesmaps** [comment]
I may do st wrong. I see this (Mapper 0.9.1):

![min](https://user-images.githubusercontent.com/20535899/74843886-de367a00-532c-11ea-8370-f6d608e9be01.jpg)

**Comment by ollesmaps** [comment]
Oh I see the problem now. It is import from Ocad. So the minimum dimension is not set. Sorry for confusion.

**Comment by valdisj** [comment]
So what is we come back to this idea once more - for legibility of map reading, quality of drawing extremely depends on if minimum sizes, dimensions and distances are taken into account. Suggestion: to implement a setting / or a process similar to OCAD legibility setting, which, when launched, does mark all the objects, if they are not according to minimum size or distance between them and others. Anybody has good simple ideas how to show that on screen?

**Comment by ollesmaps** [comment]
The _Measurement tool with built-in under minimum sizes feature_ is very good!
If there is in future a possibility to: 
1) minimize the pop up window - it occupies too much drawing spaces at the moment.
2) and still be able to see there is a minimum size clash (bright color inside the pop up window?)

I think then this issue would be solved.

**Comment by dg0yt** [comment]
@ollesmaps Did you notice that this is a dock widget which can be attached to the same region as the symbol widget? Or do you refer to the Android user interface?

**Comment by lpechacek** [comment]
On Android, the widget can be docked above, next to, or below the drawing area. Not sure if that helps Olles, though.

![image](https://user-images.githubusercontent.com/5584406/78062102-785afa80-738e-11ea-9b46-830f3f454ce7.png)

**Comment by ollesmaps** [comment]
> On Android, the widget can be docked above, next to, or below the drawing area. Not sure if that helps Olles, though.
> 
The widget is very useful! I had been using it quite extensively. The best location of the widget seems to be left down corner.

On an efficiency sidenote ;) : 

1. The widget takes up a lot of space. 
2. Most of the time the crucial info is just bellow/ over prescribed size. Or to be exact **bellow** is essential.
3. Which could be perhaps minimized into an "icon sized" widget?

OR. Could this work?
The sentence _This object is too small_ is at the second half of the widget text. You have to have the widget window rather big to see it. What about moving just a piece of this info at the beginning of the widget text. I imagine three red exclamation marks or similar. This would enable to minimize the widget window and see the crucial info at the same time.

**Comment by mlerjen** [comment]
Or an icon itself. Grayed out if no minimal size is given to this symbol. Green if size is okay, Red if not. 

X-P Maybe even adding an annoying honking sound every time the an undersized feature is drawn would save us from the upcoming 7.5K map flood  X-P

**Comment by Zerbembasqwibo** [comment]
Possibility to include live 'distance to other stuff'-control, later on?
And to start with point objects, seems easier.

But tha mappers who would need it most, will turn it off, or use other sw?


tisdag 20 juli 2021 skrev mlerjen ***@***.***>:

**Comment by ollesmaps** [comment]
Ocad introduced a simple "check directly during drawing whether the drawn line or area object corresponds to the minimum dimension". If ok - green. If not - red.

https://www.facebook.com/OCADInc/videos/1513724305797466
https://ocad.com/blog/2022/11/check-readability-of-orienteering-maps-directly/

I like the simplicity of it and direct and clear feedback when drawing.
I assume Ocad team reads it here as well :) See suggestion in this thread:
> Or it can change symbol icon background color when object is being edited - if under limit highlighted color, if ok as usuall. ![color_symbol](https://user-images.githubusercontent.com/20535899/33836133-d8924fd6-de88-11e7-8408-5a7fc83fcb81.jpg)

**Comment by dl3sdo** [comment]
What about the idea from @dg0yt of adding a special keyword to the "Find" dialog?

https://user-images.githubusercontent.com/14225666/207160561-63142965-09c1-4c4b-8b41-79d8b0c9d2bd.mp4

**Comment by dl3sdo** [comment]
> @dl3sdo Why not a tool button that does this selection (find all) inside the current window? :-)

That's also an option, in this case there will be no stepping through the findings.
